### ISSUE TITLE
helm-source-header theming doesn't update correctly

### ISSUE THREAD

**Issue by ghoot** [syl20bnr/spacemacs#4605]
(Present on `master` and `develop`)

The face styling for the helm-source-header(s) in the helm buffer are stuck with the attributes of the loaded theme the first time a helm buffer is opened. Subsequent loaded themes values are ignored.

Reproduce

1. Load spacemacs-dark
2. Observe styling on the helm-source-header (pink background etc) e.g. in `spc-b-b`
3. Load another them e.g. Zenburn
4. Observe it retains the prior themes styling for the helm source headers.
5. Themes with certain highlight line colours suffer a more pronounced effect which is apparent on single sourced helm buffers.

This is true for single and multi source helm buffers. In fact that is the source of the problem. The code added in `spacemacs-helm/packages.el` `helm-toggle-header-line` is the issue. It retains the value of whichever theme is present the first time a helm buffer is opened. 

This is the function which adjusts the look and feel for single and multiple source helm buffers. The fix is simple and retains the behaviour this code was added to address, but allows the theming to update for new themes correctly. The fix also gets a better result for the many themes which don't explicitly set helm faces.

I'm happy to raise a PR.

**Comment by nixmaniack** [comment]
How do you load theme? I use `SPC T h` (recommended way, also `SPC T n` to cycle) and seems I don't have such issue, helm source headers get the new face color of selected theme. 

![Theme Cycle](https://i.imgur.com/Ro9yasi.gif)

AFAIK, `load-theme` doesn't work nicely with spacemacs and isn't documented in docs.

**Comment by ghoot** [comment]
@nixmaniack yeah thats the method I'm testing with. On a fresh install with updated packages the issue happens on both master and develop for me.

**Comment by ghoot** [comment]
@nixmaniack are you using the theming layer ? I have that enabled with some theme tweaks (none in helm). I'll retest without that against a fresh dotspacemacs after some sleep. I have a fix done but perhaps there are other factors.

**Comment by nixmaniack** [comment]
You might want to check your custom variables. Specially `custom-set-faces` in ~/.spacemacs (or where ever your custom file is located) 

Yes, I do use theming layer with some customization.

**Comment by ghoot** [comment]
Yeah I checked those just in case any had sneaked in, I don't ever have any.

**Comment by StreakyCobra** [comment]
I have the same behaviour.

**Comment by ghoot** [comment]
I've retested this with a clean install AND the vanilla `.spacemacs` created from template, and it works fine.

However I then discovered that it can be reproduced by simply changing one option i.e. if you change `dotspacemacs-helm-no-header` from its default of `nil` to `t` then the bug presents itself.

This makes sense from what I discovered earlier and described above in the issue description. The erroneous code only gets executed IF that option is changed form the default.

I'll tidy up my fix and submit the PR.
